### PR TITLE
#1036 SMILES import: general chiral specification labels (TH, AL, SP, TB, OH ) don't work

### PR DESCRIPTION
**Pull request by even1024**

### CODE REVIEW

File: api/tests/integration/ref/basic/test_smiles.py.out [hunk 1]
```diff
@@ -121,6 +121,166 @@ c1cc2-c3c(-c2cc1)cccc3
  12  7  4  0  0  0  0
 M  END
 
+F[Po](I)(Cl)Br
```
Comment (AlexanderSavelyev) [inline api/tests/integration/ref/basic/test_smiles.py.out:124]: why don't we save information about input stereo?

File: api/tests/integration/ref/basic/test_smiles.py.out [hunk 2]
```diff
@@ -121,6 +121,166 @@ c1cc2-c3c(-c2cc1)cccc3
...
+    0.0000    0.0000    0.0000 Br  0  0  0  0  0  0  0  0  0  0  0  0
+    0.0000    0.0000    0.0000 Cl  0  0  0  0  0  0  0  0  0  0  0  0
+    0.0000    0.0000    0.0000 I   0  0  0  0  0  0  0  0  0  0  0  0
+  1  2  1  0  0  0  0
+  2  3  1  0  0  0  0
+  2  4  1  0  0  0  0
+  2  5  1  0  0  0  0
+M  END
+
+O[C@@H](C1C=CC(O)=C(O)C=1)[C@H]1CCCCN1
```
Comment (AlexanderSavelyev) [inline api/tests/integration/ref/basic/test_smiles.py.out:140]: why don't we save information about input stereo for C?

File: core/indigo-core/molecule/src/smiles_loader.cpp [hunk 3]
```diff
@@ -2931,6 +2932,32 @@ void SmilesLoader::_readAtom(Array<char>& atom_str, bool first_in_brackets, _Ato
...
                 scanner.skip(1);
             }
+            else
+            {
+                std::string current((const char*)scanner.curptr());
+                std::smatch match;
+                if (std::regex_search(current, match, std::regex("^(TH|AL)([1-2])")))
+                {
+                    atom.chirality = std::stoi(match[2]);
+                    scanner.skip(3);
```
Comment (AlexanderSavelyev) [inline core/indigo-core/molecule/src/smiles_loader.cpp:2942]: should we save with information?

File: core/indigo-core/molecule/src/smiles_loader.cpp [hunk 4]
```diff
@@ -2931,6 +2932,32 @@ void SmilesLoader::_readAtom(Array<char>& atom_str, bool first_in_brackets, _Ato
...
+                }
+                else if (std::regex_search(current, match, std::regex("^SP[1-3]")))
+                {
+                    // this type of chirality not supported. just skip it.
+                    scanner.skip(3);
+                }
+                else if (std::regex_search(current, match, std::regex(R"((TB([1-9]|1[0-9]|20)|OH([1-9]|1\d|2\d|30))(?!\d))")))
+                {
+                    const int TB_GROUP = 2;
```
Comment (AlexanderSavelyev) [inline core/indigo-core/molecule/src/smiles_loader.cpp:2951]: why this information is not saved into smiles

**Review by AlexanderSavelyev — CHANGES_REQUESTED** [review]

File: api/tests/integration/ref/basic/test_smiles.py.out (hunk 1 above)
Comment (even1024) [reply api/tests/integration/ref/basic/test_smiles.py.out:124]: Created separate ticket: https://github.com/epam/Indigo/issues/1129

**Review by AlexanderSavelyev — APPROVED** [review]

**Pull request merged**